src/jobs/jobs.controller.ts

```diff
@@ -297,150 +311,131 @@ export class JobsController {
     await Promise.all(
       jc.create.actions.map((action) => {
         return action.validate(createJobDto).catch((err) => {
+          Logger.error(err);
           if (err instanceof HttpException) {
             throw err;
           }
           throw new HttpException(
             {
               status: HttpStatus.BAD_REQUEST,
-              message: `Invalid job input. Action ${action.getActionType()} unable to validate ${
-                createJobDto.type
-              } job due to ${err}`,
+              message: `Invalid job input. Action ${action.getActionType()} unable to validate ${createJobDto.type
+                } job due to ${err}`,
             },
             HttpStatus.BAD_REQUEST,
           );
         });
       }),
     );
-  }
 
+    return jc;
+  }
 
   /**
    * Checking if user is allowed to create job according to auth field of job configuration
    */
-  // TBD
-  async instanceAuthorization(
-    jobParams: Record<string, any> | undefined,
+  async instanceAuthorizationJobCreate(
+    jobCreateDto: Record<string, any>,
     user: JWTUser,
-  ): Promise<boolean> {
-    // Accepted options:
-    // #all, #datasetOwner, #datasetOwnerOrAccess, #authenticated,
-
-    if (jobParams === undefined) {
-      // TBD
-      return true;
-    } else if (jobParams.auth == JobsAuth.All) {
-      // nothing to do here
-      return true;
-    } else if (jobParams.auth == JobsAuth.DatasetOwner) {
-      // verify that all the pids listed in the property indicated are owned by the user
-      const datasetIds = this.checkDatasetIds(jobParams);
-      const numberOfDatasets = await this.datasetsService.count({
-        where: {
-          pid: { $in: datasetIds },
-          ownerGroup: { $in: user.currentGroups },
-        },
-      });
-      const datasetsNotOwner = datasetIds.length - numberOfDatasets.count;
+    jobConfiguration: any
+  ): Promise<JobClass> {
 
-      if (datasetsNotOwner > 0) {
-        throw new HttpException(
-          {
-            status: HttpStatus.BAD_REQUEST,
-            message:
-              "Unauthorized access to " +
-              datasetsNotOwner +
-              " datasets out of " +
-              datasetIds.length,
-          },
-          HttpStatus.BAD_REQUEST,
-        );
+
+    // NOTE: We need JobClass instance because casl module works only on that. 
+    // If other fields are needed can be added later.
+    const jobInstance = new JobClass();
+    jobInstance._id = "";
+    jobInstance.ownerUser = "";
+    jobInstance.ownerGroup = "";
+    jobInstance.accessGroups = [];
+    jobInstance.requesterEmail = jobCreateDto.requesterEmail;
```

Comment:
TODO: The EmailJobAction should validate that this is defined

Reply:
We should also document that configuring an email action with `#all` could lead to potential abuse for sending spam

Reply:
Yes, we should highlight the potential security issues that entails when the #all option is used.
I already talked to our security team to have a better idea and about how to phrase it.

Reply:
we can have #all but it should have a big disclaimer in the documentation about the potential risks